### PR TITLE
Migrate to SPM for dependencies

### PR DESCRIPTION
**Pull request by Ceri-anne**
* Remove Carthage references and files
* Add Swift Package Dependencies for push-notifications-swift, NotificationBanner and ForemWebView-ios
* Amend ViewController.swift so project builds
	* Change import statement for NotificationBanner to NotificationBannerSwift
	* Add missing protocol requirements of ForemWebViewDelegate

## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Documentation Update

## Description
Migrate to Swift Package Manager from Carthage for dependencies 

## Related Tickets & Documents
[Issue#241](https://github.com/forem/DEV-ios/issues/241)

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)
N/A

## Added to documentation?

- [ ] docs.dev.to
- [x] readme
- [ ] no documentation needed

## What gif best describes this PR or how it makes you feel?

happy dancing! yay!

### CODE REVIEW

**Comment by Ceri-anne** [comment]
Currently using main branch for ForemWebView dependency as having issues trying to use verions: 
ForemWebView-dependency-issue

File: DEV-Simple.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -590,11 +558,8 @@
 				CODE_SIGN_IDENTITY = "Apple Development";
 				CODE_SIGN_STYLE = Automatic;
 				CURRENT_PROJECT_VERSION = 1;
-				DEVELOPMENT_TEAM = R9SWHSQNV8;
-				FRAMEWORK_SEARCH_PATHS = (
-					"$(inherited)",
-					"$(PROJECT_DIR)/Carthage/Build/iOS",
-				);
+				DEVELOPMENT_TEAM = 6RSSBKDAGB;
```
Comment (fdoxyz) [inline DEV-Simple.xcodeproj/project.pbxproj:561]: It's great that you've tested this locally. I'm just dropping a small reminder for when you make the PR "ready for review" to revert this change so we can maintain the original development team, thanks! 🙂

**Review by fdoxyz — COMMENTED** [review]
Thank you for taking a stab at this @Ceri-anne! Despite it being in Draft mode I gave it a try and it's working great.

> Currently using main branch for ForemWebView dependency as having issues trying to use versions 0.3 or 0.4:

Yes, using the `main` branch would work great for now 👍 

Also, could you please add `NSPhotoLibraryUsageDescription` to the `Info.plist`? That will help fix a problem when trying to add pictures when writing a post.

File: DEV-Simple.xcodeproj/project.pbxproj (hunk 1 above)
Comment (Ceri-anne) [reply DEV-Simple.xcodeproj/project.pbxproj]: I've reverted the development team change and added a `NSPhotoLibraryUsageDescription` to the info.plist 🙂

**Review by fdoxyz — APPROVED** [review]
Thank you for the quick turnaround on this @Ceri-anne!

It looks good to me, but I'm unable to merge because we have GitHub actions CI check failing.

I'm aware in this case we have a duplicate check (at some point we were looking into migrating from Travis to GH Actions entirely). Would you mind removing [these Carthage steps](https://github.com/forem/DEV-ios/blob/main/.github/workflows/ci.yml#L19-L27) from `.github/workflows/ci.yml` for now?

**Comment by Ceri-anne** [comment]
Yay! Thanks for reviewing it @fdoxyz. This was a fun task to do 😁  

I've removed those lines from `.github/workflows/ci.yml` as requested so hopefully it'll be good to go

**Comment by fdoxyz** [comment]
Awesome, thank you again @Ceri-anne!
